File: w4_array_16/doulihang/最接近的三数之和.py
#!usr/bin/env python
#-*- coding:utf-8 -*-
"""
@author:doulihang
@file: 最接近的三数之和.py
@time: 2019/10/18
"""


class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        res = 2147483647
        n = len(nums)
        for i in range(n - 1):
            left = i + 1
            right = n - 1

            while left < right:
                sum = nums[i] + nums[left] + nums[right]
                if res == target:
                    return res
                if abs(res - target) > abs(sum - target):
                    res = sum
                if sum < target:
                    left += 1
                else:
                    right -= 1
        return res

File: w4_array_16/shangyi/最接近的三数之和.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        int count = nums.length;
        int m, n, p = 0; 
        int s = 10000; 
        for(int i=0;i<count-2;i=i+1){
            for(int j=i+1;j<count-1;j=j+1){
                for(int k=j+1;k<count;k=k+1){
                    n = nums[i] + nums[j] + nums[k];
                    m = Math.abs(nums[i] + nums[j] + nums[k] - target);
                    if (m <= s) {
                        s = m;
                        p = n;
                    }
                }
            }
        }
        
        return p;
    }
}